core.endpoint.s3.s3_rest:: Could not log bucket operation: TypeError: Cannot destructure property

### Environment info
- NooBaa Version: **VERSION**
- Platform: **Kubernetes 1.14.1 | minikube 1.1.1 | OpenShift 4.1 | other: specify**

noobaa non containerized d/s rpm noobaa-core-5.17.2-20241217.el9.ppc64le 

### Actual behavior
1.

Observed in the logs the following errors , when the Warp IO workload along with the HA functionality testing when "ENABLEMD5 : true"

```
noobaa.log

Dec 18 14:41:21 node-gui0 [4081247]: 2024-12-18 14:41:21.251401 [PID-4081247/TID-4081247] [L1] FS::FSWorker::OnError: Readfile _path=/ibm/fvt_ces_root/ces
/s3-config/buckets/newbucket-warp-ha-efix-downstream-18dec2024.json  error.Message()=Protocol driver not attached
Dec 18 14:41:21 node-gui0 [4081247]: [nsfs/4081247]  [WARN] core.sdk.config_fs:: get_config_data: with config_file_path /ibm/fvt_ces_root/ces/s3-config/bu
ckets/newbucket-warp-ha-efix-downstream-18dec2024.json got an error [Error: Protocol driver not attached] { code: 'EUNATCH', context: 'Readfile _path=/ibm/fvt
_ces_root/ces/s3-config/buckets/newbucket-warp-ha-efix-downstream-18dec2024.json ' }
Dec 18 14:41:21 node-gui0 [4081247]: [nsfs/4081247] [ERROR] core.endpoint.s3.s3_rest:: Could not log bucket operation: TypeError: Cannot destructure property 'event_code' of 'undefined' as it is undefined.    at new NoobaaEvent (/usr/local/noobaa-core/src/manage_nsfs/manage_nsfs_events_utils.js:24:23)    at BucketSpaceFS.read_bucket_sdk_info (/usr/local/noobaa-core/src/sdk/bucketspace_fs.js:157:13)    at async NsfsObjectSDK._validate_bucket_namespace (/usr/local/noobaa-core/src/sdk/object_sdk.js:290:24)    at async LRUCache.get_with_cache (/usr/local/noobaa-core/src/util/lru_cache.js:67:28)    at async NsfsObjectSDK.read_bucket_sdk_config_info (/usr/local/noobaa-core/src/sdk/object_sdk.js:180:28)    at async Object.send_bucket_op_logs (/usr/local/noobaa-core/src/endpoint/s3/s3_bucket_logging.js:12:29)    at async Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:73:17)
Dec 18 14:41:21 node-gui0 [4081247]: 2024-12-18 14:41:21.321307 [PID-4081247/TID-4081275] [L0] FS::FSWorker::Execute: WARNING Readfile _path=/ibm/fvt_ces_root/ces/s3-config/buckets/newbucket-warp-ha-efix-downstream-18dec2024.json  took too long: 1029.2 ms
Dec 18 14:41:21 node-gui0 [4081247]: 2024-12-18 14:41:21.321323 [PID-4081247/TID-4081280] [L0] FS::FSWorker::Execute: WARNING Readfile _path=/ibm/fvt_ces_root/ces/s3-config/buckets/newbucket-warp-ha-efix-downstream-18dec2024.json  took too long: 1038.99 ms
Dec 18 14:41:21 node-gui0 [4081247]: 2024-12-18 14:41:21.321391 [PID-4081247/TID-4081247] [L1] FS::FSWorker::OnError: Readfile _path=/ibm/fvt_ces_root/ces/s3-config/buckets/newbucket-warp-ha-efix-downstream-18dec2024.json  error.Message()=Protocol driver not attached
Dec 18 14:41:21 node-gui0 [4081247]: [nsfs/4081247]  [WARN] core.sdk.config_fs:: get_config_data: with config_file_path /ibm/fvt_ces_root/ces/s3-config/buckets/newbucket-warp-ha-efix-downstream-18dec2024.json got an error [Error: Protocol driver not attached] { code: 'EUNATCH', context: 'Readfile _path=/ibm/fvt_ces_root/ces/s3-config/buckets/newbucket-warp-ha-efix-downstream-18dec2024.json ' }
Dec 18 14:41:21 node-gui0 [4081247]: [nsfs/4081247] [ERROR] core.endpoint.s3.s3_rest:: Could not log bucket operation: TypeError: Cannot destructure property 'event_code' of 'undefined' as it is undefined.    at new NoobaaEvent (/usr/local/noobaa-core/src/manage_nsfs/manage_nsfs_events_utils.js:24:23)    at BucketSpaceFS.read_bucket_sdk_info (/usr/local/noobaa-core/src/sdk/bucketspace_fs.js:157:13)    at async NsfsObjectSDK._validate_bucket_namespace (/usr/local/noobaa-core/src/sdk/object_sdk.js:290:24)    at async LRUCache.get_with_cache (/usr/local/noobaa-core/src/util/lru_cache.js:67:28)    at async NsfsObjectSDK.read_bucket_sdk_config_info (/usr/local/noobaa-core/src/sdk/object_sdk.js:180:28)    at async Object.send_bucket_op_logs (/usr/local/noobaa-core/src/endpoint/s3/s3_bucket_logging.js:12:29)    at async Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:73:17)
...
Dec 18 14:41:21 node-gui0 [4081247]: [nsfs/4081247] [ERROR] core.endpoint.s3.s3_rest:: Could not log bucket operation: TypeError: Cannot destructure property 'event_code' of 'undefined' as it is undefined.    at new NoobaaEvent (/usr/local/noobaa-core/src/manage_nsfs/manage_nsfs_events_utils.js:24:23)    at BucketSpaceFS.read_bucket_sdk_info (/usr/local/noobaa-core/src/sdk/bucketspace_fs.js:157:13)    at async NsfsObjectSDK._validate_bucket_namespace (/usr/local/noobaa-core/src/sdk/object_sdk.js:290:24)    at async LRUCache.get_with_cache (/usr/local/noobaa-core/src/util/lru_cache.js:67:28)    at async NsfsObjectSDK.read_bucket_sdk_config_info (/usr/local/noobaa-core/src/sdk/object_sdk.js:180:28)    at async Object.send_bucket_op_logs (/usr/local/noobaa-core/src/endpoint/s3/s3_bucket_logging.js:12:29)    at async Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:73:17)
Dec 18 14:42:30 node-gui0 node[4115777]: 2024-12-18 14:42:30.238587 [PID-4115777/TID-4115777] [L1] FS::set_log_config: LOG_TO_STDERR_ENABLED=0 LOG_TO_SYSLOG_ENABLED=1

```

It has occurred for 4 - 5 times, protocol driver UNATACH is acceptable, since during HA time , there is a movement of CES IP and at that time there are chances of trying to access these buckets in the directory might have run into this operation. 

### Expected behavior
1.
Please investigate on why the bucket operation is logged when the feature is not enabled in the environment today. 

### Steps to reproduce
1.
Ran Warp run, but it doesn't seem to be related here 

### More information - Screenshots / Logs / Other output
Box folder with link to the noobaa and mmfs.log
1) 2 times occurred on the 17th Dec noobaa logs , has different message than 18th Dec log (bucket operation message)  --> https://ibm.ent.box.com/folder/299549553376 
2) occurred mostly on 18th Dec https://ibm.ent.box.com/folder/299550817338

--- a/src/util/native_fs_utils.js
+++ b/src/util/native_fs_utils.js
@@ -660,6 +660,7 @@ function get_bucket_tmpdir_full_path(bucket_path, bucket_id) {
 /**
  * translate_error_codes we translate FS error codes to rpc_codes (strings)
  * and add the rpc_code property to the original error object
+ * default rpc_code is internal error
  * @param {object} err
  * @param {('OBJECT'|'BUCKET'|'USER'|'ACCESS_KEY')} entity
  */
@@ -671,7 +672,7 @@ function translate_error_codes(err, entity) {
     if (err.code === 'EEXIST') err.rpc_code = `${entity}_ALREADY_EXISTS`;
     if (err.code === 'EPERM' || err.code === 'EACCES') err.rpc_code = 'UNAUTHORIZED';
     if (err.code === 'IO_STREAM_ITEM_TIMEOUT') err.rpc_code = 'IO_STREAM_ITEM_TIMEOUT';
-    if (err.code === 'INTERNAL_ERROR') err.rpc_code = 'INTERNAL_ERROR';
+    if (err.code === 'INTERNAL_ERROR' || !err.rpc_code) err.rpc_code = 'INTERNAL_ERROR';
     return err;
 }
 

